### ISSUE TITLE
About data flipping

### ISSUE THREAD

**Issue by zzzendurance** [hustvl/Vim#101]
Thank you for reading my question. In the code, the data is flipped in the last dimension: xz.flip([-1]), so I want to ask if it can be changed in other dimensions (you only need to answer me this question), because I am worried that if I only change xz.flip([-1]), the code logic elsewhere is related to this code logic.

**Comment by zzzendurance** [comment]
I have just tried to flip another dimension and found that it works successfully, but I am not sure if this change is effective. If the author sees my question, thank you for your answer

**Comment by Karn3003** [comment]
Hi @zzzendurance , Can you exactly point out the line? As I do think in out_b it should be out_b.flip([1]) instead of -1. As we need to flip the tokens not the dimensions across the tokens.

**Comment by zzzendurance** [comment]
> Hi @zzzendurance , It should be 1 right? As I do think in out_b it should be out_b.flip([1]) instead of -1. As we need to flip the tokens not the dimensions across the tokens.

I think what you said makes sense. I’ve been following this code for a long time, and I almost forgot about it. I just looked it up, and it should be this line of code. This line is likely flipping the `c` dimension in the `(b, t, c)` format, right?

https://github.com/hustvl/Vim/blob/1be434d49d81f24c7d1aef3f4729d6e64a1c3c62/mamba-1p1p1/mamba_ssm/modules/mamba_simple.py#L230

**Comment by Karn3003** [comment]
> > Hi @zzzendurance , It should be 1 right? As I do think in out_b it should be out_b.flip([1]) instead of -1. As we need to flip the tokens not the dimensions across the tokens.
> 
> I think what you said makes sense. I’ve been following this code for a long time, and I almost forgot about it. I just looked it up, and it should be this line of code. This line is likely flipping the `c` dimension in the `(b, t, c)` format, right?
> 
> https://github.com/hustvl/Vim/blob/1be434d49d81f24c7d1aef3f4729d6e64a1c3c62/mamba-1p1p1/mamba_ssm/modules/mamba_simple.py#L230

Yes, it's flipping the c dimension in (b, t, c). What kind of task did you perform, and did this change work successfully??

**Comment by zzzendurance** [comment]
> > > Hi @zzzendurance , It should be 1 right? As I do think in out_b it should be out_b.flip([1]) instead of -1. As we need to flip the tokens not the dimensions across the tokens.
> > 
> > 
> > I think what you said makes sense. I’ve been following this code for a long time, and I almost forgot about it. I just looked it up, and it should be this line of code. This line is likely flipping the `c` dimension in the `(b, t, c)` format, right?
> > https://github.com/hustvl/Vim/blob/1be434d49d81f24c7d1aef3f4729d6e64a1c3c62/mamba-1p1p1/mamba_ssm/modules/mamba_simple.py#L230
> 
> Yes, it's flipping the c dimension in (b, t, c). What kind of task did you perform, and did this change work successfully??

My task is speech emotion recognition. I remember that this change worked successfully, but the effect did not seem to change much.
